sdk/python/tests/integration/feature_repos/repo_configuration.py

```diff
@@ -69,9 +70,13 @@
     IntegrationTestRepoConfig(python_feature_server=True),
 ]
 if os.getenv("FEAST_IS_LOCAL_TEST", "False") != "True":
+    # Local integration tests use local Dynamodb
+    LOCAL_DYNAMO_CONFIG = deepcopy(DYNAMO_CONFIG)
+    LOCAL_DYNAMO_CONFIG["endpoint_url"] = "http://localhost:8000"
```

Comment:
Do we set up dynamo locally somehow? We'd need to, otherwise this will fail right?

Reply:
yes, following the instructions of [Deploying DynamoDB Locally on Your Computer](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.DownloadingAndRunning.html) is a prerequisite

Reply:
@achals I've added some guidelines to `CONTRIBUTING.md` file

Reply:
@TremaMiguel I think we shouldn't make a local dynamodb installation necessity to run tests - I think it increases the bar for new contributors. It should be optional (with instructions in `CONTRIBUTING.md`) or we should use [testcontainers](https://testcontainers-python.readthedocs.io/en/latest/?badge=latest) to launch the local version of dynamodb and tear it down as necessary. 

I'm okay if you'd like to follow that up in a separate PR and just have the `endpoint_url` change in this PR.

Reply:
ok @achals , I think we should split this in two PRs, one concerning the development and testing of `endpoint_url` and the other one on how to run local integration tests with DynamoDB local as optional.

Reply:
test-containers looks an easy to implement option but we would need to add support for DynamoDB. 

I think we can add a make command `test-integration-python` to run local integration tests, for DynamoDB Online Store it will spin up a Docker Container with DynamoDB and tear it down when testing ends.

Reply:
#2493 should have example on how to spin up dynamodb-local. I think after that lands, you should be able to add a component for dynamodb as well.

Reply:
got it, in the meantime #2505 has been raised. I'll be happy to develop that once we're one with this one.